### ISSUE TITLE
Release v2.0.0

### ISSUE THREAD

**Issue by MikeAlhayek** [OrchardCMS/OrchardCore#16428]
### Prepare the project

Do some housekeeping on GitHub in the [main repo](https://github.com/OrchardCMS/OrchardCore).

- [ ] Assign the milestone of the release's version to this issue.
- [ ] Close remaining issues for the version (including merging corresponding pull requests if suitable) or assign them to the next one.
- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).

### Prepare the code and documentation

Update the source, so everything looks like on the new version.

- [ ] Create a `release/<version name>` branch (e.g., `release/1.8.0`) from `main` or the previous release branch. This is necessary to target pull requests for the upcoming release.
- [ ] Create an issue branch out of this branch as usual.
- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
- [ ] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
- [ ] Update module versions in `src/OrchardCore/OrchardCore.Abstractions/Modules/Manifest/ManifestConstants.cs`.
- [ ] Create a new milestone.
- [ ] Change the Orchard Core version in the project templates, more specifically in `dotnetcli.host.json` and `template.json` files.
- [ ] Add final updates to the release notes in the documentation. It should include the following, at least:
  - Overview of the release's highlights and goals. What do you want people to remember this release for?
  - Prerequisites. What framework version do you need, anything else to work with Orchard?
  - Upgrade steps, any migration necessary from previous versions, and any breaking changes.
- [ ] Add the release notes documentation page to the documentation site's navigation in `mkdocs.yml` and remove it from `not_in_nav`.
- [ ] Update the documentation to mention the version in all places where the latest version is referenced, for example, but not limited to (do a search for the package version string): [Status in the root README](https://docs.orchardcore.net/en/latest/#status), CLI templates, commands, the [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/) guide.

### Test the release

Make sure everything works all right.

- [ ] Make sure that [`OrchardCore.Samples` works](https://github.com/OrchardCMS/OrchardCore.Samples).
- [ ] Test the [guides](https://docs.orchardcore.net/en/latest/guides/) with the NuGet packages from the Cloudsmith feed (branches under `release/` are automatically published too). Test at least the following guides:
  - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/guides/create-modular-application-mvc/)
  - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/guides/create-cms-application/)
  - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/)
- [ ] Re-certify Orchard Core for the latest major version of Red Hat Enterprise Linux if a new version has been released (e.g., v10 after v9). Refer to Orchard's [Red Hat Ecosystem Catalog profile](https://catalog.redhat.com/software/applications/detail/223797) for the previously certified version, the [Red Hat Customer Portal](https://access.redhat.com/articles/3078) for the latest version, and [our certification guide](https://docs.orchardcore.net/en/latest/topics/red-hat-ecosystem-catalog-certification/) for the certification steps.

### Prepare and publish Orchard Core Translations

Update everything in the [Translations project](https://github.com/OrchardCMS/OrchardCore.Translations). Only do this once all the code changes are done since localized strings can change until then.

- [ ] Update .po files with [PoExtractor](https://github.com/lukaskabrt/PoExtractor). This will also update [Crowdin](https://crowdin.com/project/orchard-core).
- [ ] Publish the new version on NuGet.
- [ ] Update the `OrchardCore.Translations.All` package reference in the main repo's _./Dependencies.Packages.props_ file to refer to the new NuGet package.

### Publish the release

Do the harder parts of making the release public. This should come after everything above is done.

- [ ] Tag the head of `release/<version name>` with the version. Include `v` in the name, e.g. `v1.8.0`.
- [ ] Merge `release/<version name>` to `main`.
  - You'll need to create a pull request.
  - Merge it as a merge commit, not squash merge.
  - If there are merge conflicts, then create a `release/<version name>-integration` branch and fix them there.
- [ ] Create and publish a release [on GitHub](https://github.com/OrchardCMS/OrchardCore/releases/new).
  - Generate release notes.
  - Add a link to the release notes in the docs (something like `For details on this version see the [release notes in the documentation](link here).`). Note that the docs will only be built once the branch is merged to `main`.
- [ ] Test the [guides](https://docs.orchardcore.net/en/latest/guides/) with the packages now automatically published to NuGet. Test at least the following guides:
  - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/guides/create-modular-application-mvc/)
  - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/guides/create-cms-application/)
  - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/)
- [ ] Update [Try Orchard Core](https://github.com/OrchardCMS/TryOrchardCore).

### Publicize the release

Let the whole world know about our shiny new release. Savor this part! These steps will make the release public so only do them once everything else is ready.

- [ ] Tweet
- [ ] Post to the [Orchard Core LinkedIn group](https://www.linkedin.com/groups/13605669/)
- [ ] Post to the [Orchard Core Facebook page](https://www.facebook.com/OrchardCore/)

### After the release is done

- [ ] Create a new milestone with the next release number and close the old milestone.
- [ ] Create a new release notes documentation file for the next version in the `OrchardCore.Docs` project. (e.g., `/releases/1.8.0.md`). Don't add it to the docs navigation and exclude it from validation under `not_in_nav` with `mkdocs.yml`.
- [ ] Update the `OrchardCore.Commons.props` file with the next release number, and `<VersionSuffix>preview</VersionSuffix>` such that preview builds use the new one.

**Comment by MikeAlhayek** [comment]
We'll start this process after PR #16402 is merged.

**Comment by Piedone** [comment]
Don't do this yet, please. We had tons of regressions, the last one you discovered yesterday, so let's wait a bit to see if anything new will be discovered.

@sarahelsaig is also still working on upgrading Orchard Core Commerce (see https://github.com/OrchardCMS/OrchardCore.Commerce/pull/454) and other solutions to the latest preview. Since the goalpost keeps moving with new issues discovered, this is far from straightforward.

**Comment by MikeAlhayek** [comment]
@Piedone yes no need to release this immediately. But as far as I know that @sarahelsaig has been working on this for a while now and found one bug only "which was fixed". So I assumed there was no additional bugs found. Either way, at some point we have to draw a line and just ship so we can allow wider audience to test the new release.

**Comment by sarahelsaig** [comment]
Actually I've found 11 bugs while testing with OCC and the Lombiq modules. [The latest](https://github.com/OrchardCMS/OrchardCore/issues/16349) was fixed only 3 weeks ago. Also earlier this week I have found a new `ISession` concurrent access bug in OCC. I haven't opened an issue yet because I couldn't isolate the reproduction steps and it would not be a useful bug report without that.

**Comment by Piedone** [comment]
The point though is, having OCC and a few other large Lombiq solutions upgraded, and their over 300 UI tests passing, is a strong evidence that we're ready for a release. Otherwise, we'll have subsequent patch releases every other day.

This has been a really bumpy road, getting blocked repeatedly. Version 2 will be a very risky release, hence we had a huge amount of regressions reported. So, due to the lack of a comprehensive automated QA suite, this is our second-best bet to make sure it'll go well.

We can prevent further moving the goalposts by not merging anything else than bugfixes though.

**Comment by MikeAlhayek** [comment]
Ideally we want Lombiq projects converted and all tests pass. But, will this happen any time soon? We just don't want to hold the release indefinitely without having some sort of plan to release. Is there a target finish date for Lombiq transition?

**Comment by MikeAlhayek** [comment]
> Actually I've found 11 bugs while testing with OCC and the Lombiq modules. 

@sarahelsaig beside that ISession issue, are there any other issue from the 11 you reported that still need to be addressed?

**Comment by Piedone** [comment]
I'll defer about the timing to Sári. However, this also depends on what changes go to OC in the meantime.

**Comment by MikeAlhayek** [comment]
Once we have timeline, we'll be able to determine what will go into 2.0 and what will have to wait. But we should be looking into releasing 2.0 before August 1st as a max date. 1.8.0 was released in January 2, 2024. So we are well over due for a release.

**Comment by Piedone** [comment]
![image](https://github.com/user-attachments/assets/3387ff20-32b2-4b4e-b2fe-65b3ea707836)

**Comment by MikeAlhayek** [comment]
>  Change the Orchard Core version in the project templates, more specifically in dotnetcli.host.json and template.json files.

@Piedone I think this line item you added to the to do list. Both `dotnetcli.host.json` and `template.json` are not tracked on Github.

**Comment by Piedone** [comment]
They are part of the templates:

![image](https://github.com/user-attachments/assets/f86f3bb8-36b9-4796-b973-cc9341eba0b1)

![image](https://github.com/user-attachments/assets/288519a3-1326-48bb-91fa-9ed9dbf3fb3e)

**Comment by hyzx86** [comment]
Are there any drawbacks to gitub's nuget service?

**Comment by Piedone** [comment]
Do you mean [GitHub Packages](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry)? It isn't free, unlike Cloudsmith.

**Comment by hishamco** [comment]
@Piedone I tried a long time ago without any cost

**Comment by Piedone** [comment]
I see, it's actually free for public packages: https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages#about-billing-for-github-packages

**Comment by hishamco** [comment]
Yes it's, it might be suited for us, especially for nightly builds, the thing I suffered from at that time was referencing them, from another apps

**Comment by MikeAlhayek** [comment]
@hyzx86 @Piedone @hishamco please convert that into a discussion so we can keep this issue 2.0.0 release related only.

Thank you

**Comment by MikeAlhayek** [comment]
> They are part of the templates:
> 
> ![image](https://private-user-images.githubusercontent.com/1976647/350967851-f86f3bb8-36b9-4796-b973-cc9341eba0b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NTU4MjEsIm5iZiI6MTcyMTg1NTUyMSwicGF0aCI6Ii8xOTc2NjQ3LzM1MDk2Nzg1MS1mODZmM2JiOC0zNmI5LTQ3OTYtYjk3My1jYzkzNDFlYmEwYjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMjExMjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzBlZjY3Mzk5MzljYjU5ZWUwNjUxNjE5NTE1OGJkMmYwMWU4ZDE3MDZhOGJlZTEzZDIxZDFmN2M1ZDJlOGM3YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.SlB6D9ybNu6hCrE9fc9aIDVzix5mqxS3WOhilK1wNww)
> 
> ![image](https://private-user-images.githubusercontent.com/1976647/350967904-288519a3-1326-48bb-91fa-9ed9dbf3fb3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4NTU4MjEsIm5iZiI6MTcyMTg1NTUyMSwicGF0aCI6Ii8xOTc2NjQ3LzM1MDk2NzkwNC0yODg1MTlhMy0xMzI2LTQ4YmItOTFmYS05ZWQ5ZGJmM2ZiM2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMjExMjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQyNTJmZDk2NWFhYWEyYjQ1ZjI1MGQ0ODA1MDExZDRhMWJiZjExNDBiOTFiZTgyMmI4YWZjNTdiMDA1NjljNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2chZpu2nNsqycQoA-sLBqq5mTOW6wKwaL7r7yOPucOA)

@Piedone I think the issue is that these files are ignored from `.gitignore` I am not sure why these are ignored. @agriffard do you recall why this folder was ignore while the files are still on the server? https://github.com/OrchardCMS/OrchardCore/commit/a9122aea1a9bf4e138e63cadb86f258355f03504#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R204

In [this PR](https://github.com/OrchardCMS/OrchardCore/pull/16427/files), I removed the exclusion rule so that we can track changes to the `.template.config` folders.

**Comment by Piedone** [comment]
I don't feel like moving to GitHub Packages BTW, it seems like a pretty equivalent service.

**Comment by MikeAlhayek** [comment]
@Piedone It looks like @sarahelsaig completed the migration of e-commerce project and everything is passing. Are there any other project that we should wait for before we resume the release of 2.0?

**Comment by Piedone** [comment]
Yeah, she's working on DotNest as well, which is a different kind of test with all the random people's tenants. Note though that every breaking change like https://github.com/OrchardCMS/OrchardCore/pull/16493 necessitates another round of updates.

I'd also be a lot more confident if we can go a week without a regression being reported.

**Comment by MikeAlhayek** [comment]
> Note though that every breaking change like https://github.com/OrchardCMS/OrchardCore/pull/16493 necessitates another round of updates.

This update should not need any code change. Rebuilding the project should be all one will need. 

I am okay with waiting a week to waiting. But the question, how long will the DotNest upgrade takes? @sarahelsaig in there an rough ETA for DotNest?

**Comment by Piedone** [comment]
> This update should not need any code change. Rebuilding the project should be all one will need.

Great, then!

**Comment by sarahelsaig** [comment]
> in there an rough ETA for DotNest?

Unless there are some new surprise breaking change, I will finish it in the next couple days.

**Comment by Piedone** [comment]
@sarahelsaig can you share updates, please?

**Comment by hyzx86** [comment]
A important point is that all of our current queries `SqlQuery` , `Lucene Query` and `Graphql Query` and script extensions `executeQuery` don't support totals, so if we can't get these PRs done in the near future, can we define these api's as returning object instead of returning a collection?

**Comment by MikeAlhayek** [comment]
> The important point is that all of our current queries `SqlQuery`, `Lucene Query` and `Graphql Query` as well as the scripting extension `executeQuery` don't support totals so if we can't get these PRs done on the fly, can we start by defining these api's as returning an object rather than a collection?

@hyzx86 is this comment related to the release process of 2.0? Maybe you meant to post that on a different issue. 

This issue is strictly for releasing 2.0 not with issues that could be related to 2.0. If you have an issue with 2.0, please open a new issue and we'll tag it with 2.0 milestone.

**Comment by hyzx86** [comment]
> > The important point is that all of our current queries `SqlQuery`, `Lucene Query` and `Graphql Query` as well as the scripting extension `executeQuery` don't support totals so if we can't get these PRs done on the fly, can we start by defining these api's as returning an object rather than a collection?
> 
> @hyzx86 is this comment related to the release process of 2.0? Maybe you meant to post that on a different issue.
> 
> This issue is strictly for releasing 2.0 not with issues that could be related to 2.0. If you have an issue with 2.0, please open a new issue and we'll tag it with 2.0 milestone.

I mean these tweaks will need to introduce some disruptive updates, and since we've already done some disruptive upgrades in 2.0

**Comment by MikeAlhayek** [comment]
Then please open a separate issue with any issue/concerns you may have with the current state of main.

**Comment by sarahelsaig** [comment]
I'm close to the end now. We need this before the 2.0 release: https://github.com/OrchardCMS/OrchardCore/issues/16580
I'm also investigating another strange behavior, but I'm not sure if that's related to OC yet. I will follow up with updates later today or tomorrow.

**Comment by sarahelsaig** [comment]
@MikeAlhayek Good news, we are done testing with DotNest.

**Comment by MikeAlhayek** [comment]
@sarahelsaig that is great news! Do you want to tackle https://github.com/OrchardCMS/OrchardCore/issues/16585 ?

**Comment by sarahelsaig** [comment]
> @sarahelsaig that is great news! Do you want to tackle #16585 ?

I did it here: #16623.

**Comment by Piedone** [comment]
Nothing else remaining here before release, right?

**Comment by MikeAlhayek** [comment]
I like to take https://github.com/OrchardCMS/OrchardCore/pull/16636 in 2.0 since it has a breaking change.

**Comment by hishamco** [comment]
> Nothing else remaining here before release, right?

I need to update PO Extract tool as well

Can we expect OC `2.0.0` to be released within 1-2 weeks? If Yes I will do the update ASAP

**Comment by MikeAlhayek** [comment]
The plan is to release OC early next week. So feel free to update the translations.

**Comment by hishamco** [comment]
IMHO we could do it in the OC Harvest :) meanwhile, we can test the latest fixes and check the translations, and integrations. Also OC Commerce and DotNest could do a final check before releae

**Comment by MikeAlhayek** [comment]
We want to release it before so we can talk about it and about 3.0 during the harvest.

**Comment by hishamco** [comment]
> we can talk about it and about 3.0 during the harvest.

I'm not sure in which session we can talk about the current and future releases

**Comment by MikeAlhayek** [comment]
Probably during Q/A or side chats. I also think it's part of the keynote we are mentioning 3.0

**Comment by MikeAlhayek** [comment]
@agriffard can you please release translations package for 2.0.0?

**Comment by hishamco** [comment]
Can we hold for 1-2 days I need to bump PO Extractor update then @agriffard can release translation packages for `2.0.0`

**Comment by MikeAlhayek** [comment]
@hishamco tomorrow by noon PST, we'll need to release 2.0. I hope you'll be done by then. Since the Harvest is next week, some people will be out and/or busy. So we can't wait much longer.

**Comment by hishamco** [comment]
Let us wait for @agriffard, if the current PO Extractor version is fine I will defer bumping the new version after `2.0.0`. I don't like to harry up without proper testing

**Comment by agriffard** [comment]
I updated the translations last week: https://github.com/OrchardCMS/OrchardCore.Translations/commit/0bb805edc9c4d4ac8edca20e834cb88c11b42b1c

I checked this task:
`Update .po files with [PoExtractor](https://github.com/lukaskabrt/PoExtractor). This will also update [Crowdin](https://crowdin.com/project/orchard-core).

I am not doing anything more. 
I am not the one that usually does the 2 remaining tasks related to translations when we do a release:
```
Publish the new version on NuGet.
Update the OrchardCore.Translations.All package reference in the main repo's ./Dependencies.Packages.props file to refer to the new NuGet package. 
```

**Comment by hishamco** [comment]
@sebastienros could you please handle this or shall one of us do it?

**Comment by MikeAlhayek** [comment]
@agriffard Is this the version we should be using then? https://cloudsmith.io/~orchardcore/packages/?q=version%3A1.0.1-beta-20060

Shouldn't we call this version 2.0.0-preview to match OC versions?

**Comment by agriffard** [comment]
> @agriffard Is this the version we should be using then? https://cloudsmith.io/~orchardcore/packages/?q=version%3A1.0.1-beta-20060

I made an update with the latest changes.
Yes, The date seems to match the moment I pushed.

> Shouldn't we call this version 2.0.0-preview to match OC versions?

I don't know where this version of OC.Translations in preview is set.

**Comment by MikeAlhayek** [comment]
> @sebastienros could you please handle this or shall one of us do it?

@hishamco what are you asking here?

Are we still waiting on you to make any changes to the `OrchardCore.Translations.All` project or are the changes all done? If you are done, I can resume with the 2.0.0 release.

**Comment by hishamco** [comment]
What I mean is to publish the `OC.Translations.All`, seems @sebastienros who was doing that each time

**Comment by MikeAlhayek** [comment]
@hishamco every commit on that project generates a new translation package. I can release a production version tomorrow "if you finished everything you want to do".

**Comment by MikeAlhayek** [comment]
I am going to delay this until Monday/Tuesday. We have one report of a 2.0 bug that we are still working on and we'll give people time to test the last changes over the weekend. Any objections?

**Comment by Piedone** [comment]
Go ahead!

**Comment by hishamco** [comment]
> @hishamco every commit on that project generates a new translation package. I can release a production version tomorrow "if you finished everything you want to do".

I defer everything after `2.0.0` to avoid regression because we have limited time

**Comment by MikeAlhayek** [comment]
> Re-certify Orchard Core for the latest major version of Red Hat Enterprise Linux if a new version has been released (e.g., v10 after v9). Refer to Orchard's [Red Hat Ecosystem Catalog profile](https://catalog.redhat.com/software/applications/detail/223797) for the previously certified version, the [Red Hat Customer Portal](https://access.redhat.com/articles/3078) for the latest version, and [our certification guide](https://docs.orchardcore.net/en/latest/topics/red-hat-ecosystem-catalog-certification/) for the certification steps.

@Piedone is it time to do this yet?

**Comment by Piedone** [comment]
It's still on 9.x so I don't think so.
